opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
> Because if it's the latter and those frameworks are mixed with other instrumentation that expects auto-finish, then the only way to make this work is to have `makeActive(span) == makeActive(span, NoopObserver)`

"the only way" was a bit hasty. Another option is to let people who don't want refcounting to explicitly invoke `makeActive(span, null)`. It keeps `makeActve(span)` backwards compatible with 0.30.x and makes refcounting an opt-out feature.